src/x-platform/index.js

```diff
@@ -24,42 +25,56 @@ export function onElectron(event, callback) {
   return false
 }
 
-export function notificate({title, content, icon, slug}, callback = noop) {
-  if (macGap) {
-    macGap.notify({
-      title,
-      content,
-      sound: false
-    })
-    return
+export class Notifier {
+  constructor() {
+    if (electron) {
```

Comment:
what about my suggestion to create separate files for every platform and avoid if (electron), if (macgap) etc?